synapse/config/tls.py

```diff
@@ -295,7 +295,7 @@ def read_tls_certificate(self) -> crypto.X509:
         cert_path = self.tls_certificate_file
         logger.info("Loading TLS certificate from %s", cert_path)
         cert_pem = self.read_file(cert_path, "tls_certificate_path")
-        cert = crypto.load_certificate(crypto.FILETYPE_PEM, cert_pem)
+        cert = crypto.load_certificate(crypto.FILETYPE_PEM, cert_pem.encode())
```

Comment:
I have a feeling that `load_certificate` takes both strings and bytes, but is declared to only take bytes? Not 100% sure what's going on here.

Reply:
As long as you're meeting its contract and `cert_pem` is a `str` here then this seems perfectly reasonable.

I'm the worst offender of this but we ought to stop second-guessing type signatures (yes yes, I know they're often genuinely wrong too :P) — support for strings might be a backwards compatibility thing, for example, so this seems like the Right Thing To Do™ here.

Reply:
My guess is that it is a legacy code-path for Python 2. I'll look a bit more into it.

Reply:
It seems they both work, weirdly another place in this file we manually open the cert for binary reading. I wonder if that would make mores sense to do here.

Reply:
> It seems they both work, weirdly another place in this file we manually open the cert for binary reading. I wonder if that would make mores sense to do here.

well now that you mention it... yeah that could make sense rather than decoding and reencoding :)

Reply:
Hmm, OK. I might revert this for now then and handle this in a future PR.